lib/src/component_declaration/component_base.dart

```diff
@@ -124,6 +124,16 @@ abstract class UiComponent<TProps extends UiProps> extends react.Component {
     );
   }
 
+  /// Validation method called during [componentWillReceiveProps], and [componentWillMount].
+  ///
+  /// Use this as an opportunity validate props during the correct lifecycle of the component.
+  @mustCallSuper
+  void validateProps([Map appliedProps]) {
```

Comment:
I think this should be a required positional parameter, so that consumers don't have to null-coalesce the argument in overrides.